upgrades/upgrade_testing_framework/steps/step_report_results.py

```diff
@@ -0,0 +1,115 @@
+import xml.etree.ElementTree as ET
+from typing import Dict, List
+from pathlib import Path
+from itertools import groupby
+import textwrap
+
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+from upgrade_testing_framework.core.expectation import load_knowledge_base, Expectation
+
+
+class ReportResults(FrameworkStep):
```

Comment:
There's quite a bit of library code in this Step file that should be moved to the `core` module and unit tested.